### PR TITLE
Species-restricted jobs + new "Trader" job for vox

### LINKED ISSUE #4796: DJ Station Sleeper has a console
It's supposed to be a non-functional sleeper, after the medcondecon update it now has a console that is inside the window it's against.
Happened in Taxi, untested in others

### LINKED ISSUE #9052: Job availability dependent on species selection
If there were some way we could have it that job selection is dependent on your species, not only could we bring in some of the more forgotten species that are still in the coderbase (Unathi would be confined to service and the janitorial duties, etc.) , and make it so you can't have, say, a diona as captain which wouldn't make much sense on a station which is run with a bias towards humans, but could also have unique jobs for said species so as to make it easier for them to be spawned elsewhere.

This would mean that you could have, say, vox could have 'Vox trader' or 'Vox miner' or 'Vox botanist' etc. that would put them in the vox outpost, plasmaman could have 'derelict survivor' that would put them somewhere on the derelict, greys or skrells could have '[species] doctor' that would put them on the medical shuttle.

Just something to give a bit more customization, and finally give some use to the vox outpost besides a landing zone for the vox shuttle

### PR DESCRIPTION
**Pull request by unid15**
[Ready for NFD]

Resolves #9052
Fixes #4796
- Jobs can now have a species blacklist or a species whitelist (so you can make the captain job unavailable to dionae, or add a job that only vox can have). They can also have a map blacklist/whitelist (so some jobs can only appear on some maps)
- Added a Trader job, only available to roundstart vox (alt titles: Merchant, Vagabond, Traveler). Traders start on the vox outpost and can do whatever they want in there.
- Vox outpost now starts with 20 metal and glass, water/fuel tanks, emergency/mechanical/electrical toolboxes, 3 station bounced radios, 1 headset, paper bins and pens, a bucket, mop and circuit boards for autolathe, washing machine, microwave and arcade
- Traders are not on the crew manifest, they don't spawn with any money, ID, PDA or even a headset.
- There can only be 3 traders
- Because there is no vox outpost on metaclub, metaclub traders spawn on a shuttle that crashed into the asteroid. The shuttle comes with 50 metal and glass, 3 pickaxes, station bounced radios, 3 sleepers, defibrillator, an emergency toolbox, a dead vox and some lanterns. The shuttle is located between the mining outpost and the research outpost, and spawns even if there are no vox traders.
- Metaclub vox traders also get a different starting message: _"A while ago you got your equipment together and boarded a small shuttle, heading for your destination (whatever it may be). Everything was great, until the shuttle somehow crashed into an asteroid. You're still alive, but you're not quite sure where you are. Maybe some of your friends know - unless they died in the crash..."_
- Also made some slight improvements to the job code. It's still a big mess

### CODE REVIEW

**Comment by gbasood** [comment]
I mean fuck at least put 'resolves #9052' in the OP if you're going to do this

**Comment by unid15** [comment]
Chill, I was just about to do that

File: code/datums/datacore.dm [hunk 1]
```diff
@@ -46,6 +46,12 @@
 		else
 			assignment = "Unassigned"
 
+		var/datum/job/job = job_master.GetJob(H.job)
+		if(!job)
+			world << "o shit ([H.job])"
```
Comment (PJB3005) [inline code/datums/datacore.dm]: Watch out!

File: code/game/jobs/job/species_unique.dm [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	no_id = 1
+
+/datum/job/trader/equip(var/mob/living/carbon/human/H)
+	if(!H)	return 0
+	H.equip_or_collect(new /obj/item/clothing/under/vox/vox_robes(H), slot_w_uniform)
+	H.equip_or_collect(new /obj/item/clothing/shoes/magboots/vox(H), slot_shoes)
+	H.equip_or_collect(new H.species.survival_gear(H.back), slot_in_backpack)
+	return 1
+
+/datum/job/trader/introduce(mob/M, job_title)
```
Comment (gbasood) [inline code/game/jobs/job/species_unique.dm]: Does this really  need to exist, since it seems to be exactly the same as its parent?

Comment (unid15) [inline code/game/jobs/job/species_unique.dm]: It's different

> You've finally got your equipment together, such as it is. Now it's time for action and adventure! In the rush of excitement, you've forgotten where you were going to go. If only you had any friends that could remind you...

vs

> As the [job_title] you answer directly to [src.supervisors]. Special circumstances may change this.

**Comment by FudgePucker** [comment]
:+1:
Also restrict non humans from being captain.
A: to prevent 'ai law two kill the captain'
B: fuck xenoscum, nt wouldn't put a fucking bird or tree in the highest command position

**Comment by AutismSpider** [comment]
Just white list Head rolls for human only all the other jobs are fine for humans and non humans etc

**Comment by ihadtoregisterforthis** [comment]
:+1: to the possibility to whitelist races _(doing jobs)_
:+1: to vox outpost roundstart

:-1: "just remove all nonhumans from command already"

**Comment by Intigracy** [comment]
I think just captain should be blacklisted.

**Comment by CptWad** [comment]
if anything everything but assistant and librarian should be blacklisted for catbeasts

**Comment by Intigracy** [comment]
You can't be a roundstart catbeast, this is a nonissue and waste of time.

**Comment by Probe1** [comment]
ban vox from head roles when

 space racism, ho!

**Comment by Exxion** [comment]
I thought we already had the whitelist certain species for certain jobs feature
At the very least, there's a config file for it

**Comment by Intigracy** [comment]
That's the config for whitelisting species themselves.

**Comment by Exxion** [comment]
No, there's also one for species-job combinations
Currently the only entry in the example config whitelists unathi for janitor or something

**Comment by unid15** [comment]
@Exxion, that's slightly different - it only allows you to select a species based on your job, whereas this allows you to select a job based on your species. Besides, I'm not sure if that even works

Anyways, metaclub traders now spawn on this shuttle . It's completely broken and dark, but there are some supplies that they can use to survive. It's located between the mining outpost, the science outpost and the outpost overrun by xenos. Other maps were also updated, now the vox outposts have some toolboxes, pickaxes and station bounced radios

**Comment by Exxion** [comment]
Won't those sleepers generate consoles?

**Comment by unid15** [comment]
nope I made it so that they won't

**Comment by Intigracy** [comment]
#4796 pls

**Comment by NigglyWiggly** [comment]
@CptWad 

> if anything everything but assistant and librarian should be blacklisted for catbeasts. EDIT: Maybe clown too.
> Librarian

No thank you

**Comment by Intigracy** [comment]
:+1: love ya unid

**Comment by Ephx1** [comment]
This is the coolest thing since buttered toast and now I want to play again.
:+1: 

Although I still would want Vox denied from head roles automatically, but that's an admin thing, so no worries there.

**Comment by gbasood** [comment]
Not sure how I feel about giving players an off-station job (vox trader) that encourages them to dick around in space

Especially since they have NOTHING TO DO on trade post besides try to coax miners into visiting and buying their bird poop in exchange for diamonds

Although I think it'd be funny (probably not viable) to make them speak only Vox and Tradeband.

**Comment by FudgePucker** [comment]
That just means they won't take an important job just to dick around in space.

**Comment by gbasood** [comment]
True I guess
I AM interested in off-station roles in general though. In particular, roles in contention with the main station (such as SYNDIE STATION which will never happen for mostly good reasons) although maybe those should be saved for gamemodes

**Comment by SysadminJeroen** [comment]
Maybe get the Vox traders to be some sort of Silk Road where they can get
certain items if they deliver it to Poxcomm or whatever.
Certain items will have a small probability of getting asked and some items
are pretty much always on the list, but their quantities change every new
round.

2016-03-29 18:44 GMT+02:00 sood notifications@github.com:

**Comment by gbasood** [comment]
So, shady cargo?

**Comment by ghost** [comment]
All vox as traders WHEN

**Comment by gbasood** [comment]
@DrCelt ew

**Comment by ghost** [comment]
@gbasood nice try vox sympathiser scum

**Comment by ghost** [comment]
Acceptable races:
Human
Grey
OK I GUESS if maybe a little OP tier: 
Plasmamen
Diona
Literally Furry Bait:
Vox
Tajaran
How do you get worse than the above tier:
Skellingtons

**Comment by SysadminJeroen** [comment]
@gbasood Could make it so only comidities can be purchased from _human_ Cargo, and anything luxurious or exotic from Vox Traders. So that might mean to remove that machine that vends wallets, snap-pops, boombox and beachball and move it to Vox.

Maybe some list of things you can purchase at Vox for credits, or which you can trade if you have $(insert_resources) sent from a Vox cargo ship.

**Comment by Intigracy** [comment]
Or he can just leave it how it is where they have starting supplies and no goals.

Jesus christ not everything has to be overly complicated.

**Comment by SysadminJeroen** [comment]
@Intigracy Wouldn't that be more like Assistants instead of Traders?

**Comment by Probe1** [comment]
It would make a nice place to shoehorn things that don't really work well but are in the game.  Like that console in cargo that gives food.

**Comment by gbasood** [comment]
@Intigracy just trying to think of how to make the role itself FUN and not just 'fuck the fact that you're a  trader and explore vaults all round'

**Comment by CptWad** [comment]
@gbasood id say have it so they have to set up their own power and atmos, maybe an AME with limited fuel tanks

**Comment by ghost** [comment]
conflicts resolved and merged when?

**Comment by gbasood** [comment]
I mean shit if we leave the trading station and its roles as-is I feel like the job might have to become whitelist only due to shittery

**Comment by Exxion** [comment]
Wouldn't that be a config thing rather than a code thing?

**Comment by gbasood** [comment]
You'd need to code the option to config it in the first place I'd believe?
Not sure how job whitelisting works as is

**Comment by Probe1** [comment]
please tag map PRs as the map they effect @gbasood @PJB3005 @clusterfack

**Comment by Probe1** [comment]
This is too good to die.

**Comment by unid15** [comment]
Will fix conflicts tomorrow

**Comment by Probe1** [comment]
yay unids da bess

**Comment by Brolaire** [comment]
merge when?

**Comment by unid15** [comment]
Conflicts fixed, also added some supplies to the vox outpost (added circuit boards for autolathes, washing machines, arcades and microwaves, 20 metal/glass, paper and pens, water and fuel tanks).

**Comment by Probe1** [comment]
merge when
(even though it'll fuck up my PR and I don't know how to fix conflicts on maps.  merge when)

**Comment by Exxion** [comment]
We have a .dmm merge engine, just Travis doesn't know how to use it
Simply updating your branch from the main one will _usually_ fix map conflicts
It doesn't work if the same individual tile was changed in both branches though

**Comment by unid15** [comment]
The map merger has almost never worked for me

I just copy my changes somewhere (to a different map), load the new version of the map and paste the changes into it, unless the change is trivial.

**Comment by Exxion** [comment]
Perhaps it just sucks ass
I wonder if TG or someone has a better one

**Comment by PJB3005** [comment]
@Exxion nope, theirs doesn't support merging conflicting maps at all.

**Comment by Exxion** [comment]
Oh, apparently, they don't need to, because they use a map file structure that Git itself works better with.

**Comment by Probe1** [comment]
PULL THE TRIGGER

ITS TIME FOR THIS BABY TO LAUNCH

**Comment by SysadminJeroen** [comment]
# MY BODY IS READY

**Comment by clusterfack** [comment]
Please make the trader existing as a job softcoded onto the map datum

**Comment by Intigracy** [comment]
@unid15 did you install the map merge driver in your .git/config file?

```
Merging
The easiest way to do merging is to install the merge driver. For this, open `vgstation13/.git/config` in a text editor, and paste the following lines to the end of it:

[merge "merge-dmm"]
    name = mapmerge driver
    driver = ./maptools/mapmerge.sh %O %A %B

After this, merging maps should happen automagically unless there are conflicts(a tile that both you and someone else changed). If there are conflicts, you will unfortunately still be stuck with opening both versions in a map editor, and manually resolving the issues.
```

**Comment by ihadtoregisterforthis** [comment]
# MY BODY IS ALSO READY

**Comment by Konater** [comment]
WHEN?

**Comment by CptWad** [comment]
Merge he

**Comment by unid15** [comment]
Oh man I finally got a week off

@clusterfack Added map black/whitelists for jobs, made the trader job only available on box/meta/deff/taxi (it won't show up on other maps at all). Should be good to go!

**Comment by SysadminJeroen** [comment]
IT MERGED!!

**Comment by Exxion** [comment]
HAPPENING

Maybe I'll go break out my really obnoxious old vox character for a round or two

**Comment by CptWad** [comment]
OH YEAH

**Comment by SkowronX** [comment]
YOU BROKE EVERYTHING

**Comment by icantthinkofanameritenow** [comment]
I told you all this would fuck everything over, but nooo. You insisted in pushing the meme trader anyway. You deserved this. 
image

**Comment by clusterfack** [comment]
You have to make a migration for this to work properly under a new sqlite schema
https://github.com/d3athrow/vgstation13/pull/2763/files

**Comment by icantthinkofanameritenow** [comment]
Let's not even bother, the one round it worked the "traders" just acted like botanists with even less purpose. It's just not a productive addition to the game.

**Comment by Probe1** [comment]
Two of the three people who were playing reported they had a fun time and would try it again.  I didn't ask the third.

**Comment by clusterfack** [comment]
Well vox players are delusional so that's understandable

**Comment by Intigracy** [comment]
rude

**Pull request merged**